### ISSUE TITLE
Disallow infix macros?

### ISSUE THREAD

**Issue by c42f** [JuliaLang/julia#44532]
While rewriting the Julia parser in [JuliaSyntax.jl](https://github.com/c42f/JuliaSyntax.jl) I've discovered a few weird things. One of the weirdest is infix macros:

```julia
julia> macro +(a,b)
           QuoteNode((a,b))
       end
@+ (macro with 1 method)

julia> @(x + y*2)
(:x, :(y * 2))
```

If I understand correctly, this is a bug due to the way the parser uses `parse-atom` inside `parse-macro-name` which accepts more than just identifiers. Then ends up using `macroify-call` to rewrite the call to `+` into a macro call.

Was this is an accident? I think it's pretty weird and we should disallow it!

**Comment by JeffBezanson** [comment]
Yes let's disallow it.

**Comment by vtjnash** [comment]
IIUC, this is not quite just macro-related, but somehow an interaction with `@doc` parsing, since it is already illegal, except if the macro appeared in doc-string location (which causes us to do a bit more look ahead).

**Comment by LilithHafner** [comment]
This is fixed in 1.10
